### ISSUE TITLE
Error when loading model

### ISSUE THREAD

**Issue by mgrenander** [mohismist/Summarization#1]
Hi,

I am looking to run the model. I get the following error when running in eval mode:
```
INFO: Loading checkpoint logs/train/model.ckpt-109924
INFO:tensorflow:Restoring parameters from logs/train/model.ckpt-109924
2018-05-08 16:49:45.028981: W tensorflow/core/framework/op_kernel.cc:1202] OP_REQUIRES failed at save_restore_v2_ops.cc:184 : Not found: Key seq2seq/decoder/attention_decoder/lstm_cell/bias not found in checkpoint
ERROR:tensorflow:Failed to restore checkpoint: logs/train, sleep for 10 secs
```
Any idea why the model cannot be loaded? Should I train it from scratch?

Many thanks!

**Comment by zwc12** [comment]
@mgrenander I guess that has sth to do with the tensorflow version. For contrast, I post my results here with tensorflow 1.30, python 2.7
```
~/Projects/NLP/seq2seq$ python summary.py --mode=eval --data_path=bin/eval_0.bin 
INFO: Starting seq2seq_attention model in eval mode...
INFO: Finished reading 40000 of 40000 words in vocab, last word added: stroller
INFO: Building graph...
INFO: Adding attention_decoder of 30 timesteps...
INFO: Time to build graph: 5.86625599861 seconds
2018-05-11 11:10:26.822980: W tensorflow/core/platform/cpu_feature_guard.cc:45] The TensorFlow library wasn't compiled to use SSE4.1 instructions, but these are available on your machine and could speed up CPU computations.
2018-05-11 11:10:26.823009: W tensorflow/core/platform/cpu_feature_guard.cc:45] The TensorFlow library wasn't compiled to use SSE4.2 instructions, but these are available on your machine and could speed up CPU computations.
2018-05-11 11:10:26.823017: W tensorflow/core/platform/cpu_feature_guard.cc:45] The TensorFlow library wasn't compiled to use AVX instructions, but these are available on your machine and could speed up CPU computations.
2018-05-11 11:10:26.823023: W tensorflow/core/platform/cpu_feature_guard.cc:45] The TensorFlow library wasn't compiled to use AVX2 instructions, but these are available on your machine and could speed up CPU computations.
2018-05-11 11:10:26.823029: W tensorflow/core/platform/cpu_feature_guard.cc:45] The TensorFlow library wasn't compiled to use FMA instructions, but these are available on your machine and could speed up CPU computations.
2018-05-11 11:10:26.933757: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:901] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2018-05-11 11:10:26.934149: I tensorflow/core/common_runtime/gpu/gpu_device.cc:887] Found device 0 with properties: 
name: GeForce 940MX
major: 5 minor: 0 memoryClockRate (GHz) 1.2415
pciBusID 0000:02:00.0
Total memory: 1.96GiB
Free memory: 1.69GiB
2018-05-11 11:10:26.934171: I tensorflow/core/common_runtime/gpu/gpu_device.cc:908] DMA: 0 
2018-05-11 11:10:26.934179: I tensorflow/core/common_runtime/gpu/gpu_device.cc:918] 0:   Y 
2018-05-11 11:10:26.934187: I tensorflow/core/common_runtime/gpu/gpu_device.cc:977] Creating TensorFlow device (/gpu:0) -> (device: 0, name: GeForce 940MX, pci bus id: 0000:02:00.0)
INFO: Loading checkpoint logs/train/model.ckpt-82640
INFO:tensorflow:Restoring parameters from logs/train/model.ckpt-82640
INFO: seconds for eval step: 1.26974892616
INFO:tensorflow:loss: 1.439236
INFO: Found new best model with 1.43923616409 running_avg_loss. Saving to logs/eval/best
INFO: Loading checkpoint logs/train/model.ckpt-82640
INFO:tensorflow:Restoring parameters from logs/train/model.ckpt-82640
INFO: seconds for eval step: 0.75452709198
INFO:tensorflow:loss: 1.326028
INFO: Found new best model with 1.43810408115 running_avg_loss. Saving to logs/eval/best
INFO: Loading checkpoint logs/train/model.ckpt-82640
INFO:tensorflow:Restoring parameters from logs/train/model.ckpt-82640
INFO: seconds for eval step: 0.751827955246
INFO:tensorflow:loss: 1.059722
```
it seem ok.
so maybe you should try it on another tensorflow environment or just train the model from scratch.

**Comment by mgrenander** [comment]
Hi @weichengzhang you were right, the tensorflow version I was using was incompatible. However, for reason 1.3 would not work for me, but **1.0** worked fine.

The issue I was seeing is related to this:
<https://github.com/KranthiGV/Pretrained-Show-and-Tell-model/issues/7#issuecomment-309862894>

I can't really explain why 1.3 worked for you and not me but the model is working for me now. Thank you for your help!

**Comment by zwc12** [comment]
@mgrenander 
The issue you mentioned answers the question: after version 1.20, parameter names of lstm cells have been changed. So the model trained with older version could not be loaded.
That means v1.30 shouldn't work, maybe I got it wrong. I'll check it out later, thanks for your asking!

**Comment by mgrenander** [comment]
Makes sense! I'll close the issue.

**Comment by zwc12** [comment]
Update: 
the results I posted was runing with tensorflow 1.10, python 2.7, not 1.30, sorry for that mistake.
